cdf_fabric_replicator/data_modeling.py

```diff
@@ -269,8 +274,9 @@ def get_instances(
 
     def get_instances_from_result(
         self, instances_from_result: NodeList | EdgeList
-    ) -> Dict[str, Any]:
+    ) -> tuple[Dict[str, Any], Dict[str, Any]]:
         instances = {}
+        deleted_instances = {}
```

Comment:
Here you can use a defaultdict instead, i.e
```
deleted_instances = defaultdict(list)
```

This way you don't have to do the "if table_name not in deleted_instances" further down . Same goes for instances. It will simply default to make a list if the key is not present yet.